ansible_events/key.py

```diff
@@ -0,0 +1,39 @@
+import asyncio
+import logging
+import os
+import shutil
+import stat
+import tempfile
+
+logger = logging.getLogger()
+
+ssh_add = shutil.which("ssh-add")
+
+
+async def install_private_key(private_key):
+
+    local_working_directory = tempfile.mkdtemp()
```

Comment:
Better use `tempfile.TemporaryDirectory` that provides a context manager to ensure that the directory is deleted after the process is finished.